UOP1_Project/Assets/Scripts/Characters/Character.cs

```diff
@@ -1,165 +1,187 @@
-	{
-		// if player has to slide then add sideways speed to make it go down
-		if (shouldSlide)
-		{
-			movementVector.x += (1f - hitNormal.y) * hitNormal.x * (speed - slideFriction);
-			movementVector.z += (1f - hitNormal.y) * hitNormal.z * (speed - slideFriction);
+				// The character is either jumping or in freefall, so gravity will add up
+				gravityContributionMultiplier = Mathf.Clamp01(gravityContributionMultiplier);
+				verticalMovement += Physics.gravity.y * gravityMultiplier * Time.deltaTime * gravityContributionMultiplier; // Add gravity contribution
+
+				// Note that even if it's added, the above value is negative due to Physics.gravity.y
+				// Cap the maximum so the player doesn't reach incredible speeds when freefalling from high positions
+				verticalMovement = Mathf.Clamp(verticalMovement, -maxFallSpeed, 100f);
+			}
+			else
+			{
+				// Full speed ground movement
+				movementVector = inputVector * speed;
+
+				// Resets the verticalMovement while on the ground,
+				// so that regardless of whether the player landed from a high fall or not,
+				// if they drop off a platform they will always start with the same verticalMovement.
+				// -5f is a good value to make it so the player also sticks to uneven terrain/bumps without floating
+				if (!isJumping)
+				{
+					verticalMovement = fallingVerticalMovement;
+					gravityContributionMultiplier = 0f;
+				}
+			}
+
+			UpdateSlide();
+
+			// Apply the result and move the character in space
+			movementVector.y = verticalMovement;
+			characterController.Move(movementVector * Time.deltaTime);
+
+			// Rotate to the movement direction
+			movementVector.y = 0f;
+			if (movementVector.sqrMagnitude >= ROTATION_TRESHOLD)
+			{
+				float targetRotation = Mathf.Atan2(movementVector.x, movementVector.z) * Mathf.Rad2Deg;
+				transform.eulerAngles = Vector3.up * Mathf.SmoothDampAngle(
+					transform.eulerAngles.y,
+					targetRotation,
+					ref turnSmoothSpeed,
+					turnSmoothTime);
+			}
 		}
-		// check if the controller is grounded and above slope limit
-		// if player is grounded and above slope limit
-		// player has to slide
-		if (characterController.isGrounded)
+
+		private void OnControllerColliderHit(ControllerColliderHit hit)
 		{
-			currentSlope = Vector3.Angle(Vector3.up, hitNormal);
-			shouldSlide = currentSlope >= characterController.slopeLimit;
+			hitNormal = hit.normal;
+			bool isMovingUpwards = verticalMovement > 0f;
+			if (isMovingUpwards)
+			{
+				// Making sure the collision is near the top of the head
+				float permittedDistance = characterController.radius / 2f;
+				float topPositionY = transform.position.y + characterController.height;
+				float distance = Mathf.Abs(hit.point.y - topPositionY);
+				if (distance <= permittedDistance)
+				{
+					// Stopping any upwards movement
+					// and having the player fall back down
+					isJumping = false;
+					gravityContributionMultiplier = 1f;
+					verticalMovement = 0f;
+				}
+			}
```

Comment:
These changes look incorrect to me, there's actual code change happening here instead of just styling changes

Reply:
We were missing `namespace`, so I added such. This adds aditional 4tabs.

Reply:
I don't think adding namespaces would make it change like this, it feels like things have been moved around which makes it nearly impossible to review and ensure that the code is still the same as before just with different styling

Reply:
I'm starting to see the changes now, damn Github is bad at displaying this..